dev-python/clade/clade-3.3.7.ebuild

```diff
@@ -0,0 +1,26 @@
+# Copyright 2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{7..9} )
+DISTUTILS_USE_SETUPTOOLS=rdepend
+
+inherit distutils-r1
+
+DESCRIPTION="Tool for extracting information about software build process and source code"
+HOMEPAGE="https://pypi.org/project/clade/"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64"
+
+RDEPEND="
+	dev-python/ujson[${PYTHON_USEDEP}]
+	dev-python/chardet[${PYTHON_USEDEP}]
+	dev-python/cchardet[${PYTHON_USEDEP}]
+	dev-python/graphviz[${PYTHON_USEDEP}]
+	dev-python/ply[${PYTHON_USEDEP}]
+"
+BDEPEND="dev-util/cmake"
```

Comment:
Tests too, [please](https://dev.gentoo.org/~mgorny/python-guide/distutils.html#enabling-tests)! Also, if we're using cmake, do we need to be using the eclass to pick up the goodness required (it does some special stuff to adapt to Gentoo)?

Reply:
> Tests too, please!

Tests will not work. They require additional external components to download.

> Also, if we're using cmake, do we need to be using the eclass to pick up the goodness required

I think no. setup.py script calls cmake internally (https://github.com/17451k/clade/blob/master/setup.py#L61-L67) to build C-libs for intercepting.

Reply:
In that case, we normally ask upstream to make downloading optional / allow local copies, and try to download manually in SRC_URI conditionally based on test?. But if it's too much work to do that, we just ask them to make it optional/allow local copies.

Reply:
I already asked the author to make tests that depends on external components conditional.